src/CMakeLists.txt

```diff
@@ -330,20 +330,20 @@ option(WITH_SYSTEM_FMT "build with system fmt" NO)
 if(WITH_SYSTEM_FMT)
   find_package(fmt 9.1.0...<10.0.0 REQUIRED)
 endif()
+if (WITH_FMT_HEADER_ONLY)
+  message(STATUS "Using fmt header-only.")
+  set(FMT_LIB fmt::fmt-header-only)
+else()
+  message(STATUS "Linking to fmt library.")
+  set(FMT_LIB fmt::fmt)
+endif()
 if(fmt_FOUND)
   message(STATUS "Building with system fmt.")
-  if (WITH_FMT_HEADER_ONLY)
```

Comment:
does this address an problem which already exist before this change or a regression introduced by the first commit?

Reply:
One that existed before. When I ran ldd on the compiled binaries when I did build with header only specified on a system version of libfmt 9, I was seeing the shared libfmt. And when watching the verbose compile with the submodule I saw it linking to the static library.